### PR TITLE
chore(Dialog): migrate to new motion APIs

### LINKED ISSUE #30041: [Bug]: Dialog animation is not respecting reduced motion OS setting
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://react.fluentui.dev/?path=/docs/components-dialog--default
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-dialog--default

### Bug Description

Use reduced motion:
image

## Actual Behavior
Animation is enabled when Dialog appears

## Expected Behavior
Dialog is shown without animation

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #30695: [chore]: migrate Dialog to use @fluentui/react-motions-preview
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

no

### Reproduction

N/A

## Actual Behavior

Uses motion defined by CSS & `react-transition-group`.

## Expected Behavior

Uses a motion defined in `@fluentui/react-motions-preview`.

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by layershifter**
## Previous Behavior

`Dialog` uses `react-transition-group` for animations.

## New Behavior

`Dialog` uses `@fluentui/react-motions-preview` for animations.

## Related Issue(s)

Fixes #30041.
Fixes #30695.
Fixes https://github.com/microsoft/fluentui/issues/31369

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Avatar-mount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Avatar-mount.html">638</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Button-mount_base.html">300</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Button-mount.html">315</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Field-mount_base.html">1149</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Field-mount.html">1201</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProvider-mount_base.html">733</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProvider-mount.html">714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-mount.html">89</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">32</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">35</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/MakeStyles-mount_base.html">879</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/MakeStyles-mount.html">896</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Persona-mount_base.html">1793</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/Persona-mount.html">1710</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/SpinButton-mount_base.html">1362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/SpinButton-mount.html">1426</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/SwatchPicker-mount_base.html">1556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31380/perf-test-react-components/SwatchPicker-mount.html">1550</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.139 MB`<br />`275.438 kB` | `1.141 MB`<br />`276.102 kB` | `1.788 kB` <br />`664 B` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `100.192 kB`<br />`29.884 kB` | `99.327 kB`<br />`29.749 kB` | `-865 B` <br />`-135 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.55 kB`<br />`20.584 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `221.825 kB`<br />`62.568 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.037 kB`<br />`14.418 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6e7021a53c286d6e7e5e43b5f9bad626799700ec'>6e7021a53c286d6e7e5e43b5f9bad626799700ec</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31380&commitId=e46ee72aa3eadf849b90af76219a7d213e60d6e9&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|5| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Dialog</b> 47 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Dialog.Fluid Actions Start.chromium.png|1267| Changed|
|Dialog.actions position start & position end - Dark Mode.chromium.png|3622| Changed|
|Dialog.actions position start.chromium.png|1267| Changed|
|Dialog.alert - RTL.chromium.png|1267| Changed|
|Dialog.alert - High Contrast.chromium.png|3622| Changed|
|Dialog.Fluid Actions End.chromium.png|1267| Changed|
|Dialog.alert - Dark Mode.chromium.png|3622| Changed|
|Dialog.actions position start & position end - High Contrast.chromium.png|3622| Changed|
|Dialog.actions position start - Dark Mode.chromium.png|3622| Changed|
|Dialog.Integration Combobox Inline.chromium.png|738| Changed|
|Dialog.actions position start & position end.chromium.png|1267| Changed|
|Dialog.actions position start - High Contrast.chromium.png|3622| Changed|
|Dialog.actions position start - RTL.chromium.png|1267| Changed|
|Dialog.actions position start & position end - RTL.chromium.png|1267| Changed|
|Dialog.alert.chromium.png|1267| Changed|
|Dialog.default - Dark Mode.chromium.png|3622| Changed|
|Dialog.default.chromium.png|1267| Changed|
|Dialog.default - RTL.chromium.png|1267| Changed|
|Dialog.nested - Dark Mode.chromium.png|6066| Changed|
|Dialog.default - High Contrast.chromium.png|3622| Changed|
|Dialog.nested - High Contrast.chromium.png|6067| Changed|
|Dialog.no actions - RTL.chromium.png|1240| Changed|
|Dialog.no actions.chromium.png|1240| Changed|
|Dialog.no actions - Dark Mode.chromium.png|3494| Changed|
|Dialog.no title & no actions - High Contrast.chromium.png|3405| Changed|
|Dialog.nested.chromium.png|2732| Changed|
|Dialog.no title & no actions - RTL.chromium.png|1220| Changed|
|Dialog.nested - RTL.chromium.png|2732| Changed|
|Dialog.no title - Dark Mode.chromium.png|3532| Changed|
|Dialog.no title & no actions - Dark Mode.chromium.png|3405| Changed|
|Dialog.no title & no actions.chromium.png|1220| Changed|
|Dialog.no title - RTL.chromium.png|1248| Changed|
|Dialog.no actions - High Contrast.chromium.png|3494| Changed|
|Dialog.no title.chromium.png|1248| Changed|
|Dialog.no title - High Contrast.chromium.png|3533| Changed|
|Dialog.non-modal - RTL.chromium.png|2700| Changed|
|Dialog.non-modal - High Contrast.chromium.png|6511| Changed|
|Dialog.non-modal - Dark Mode.chromium.png|6510| Changed|
|Dialog.scroll long content - High Contrast.chromium.png|2895| Changed|
|Dialog.non-modal.chromium.png|2700| Changed|
|Dialog.scroll long content.chromium.png|611| Changed|
|Dialog.scroll long content - Dark Mode.chromium.png|2894| Changed|
|Dialog.title custom action - Dark Mode.chromium.png|3507| Changed|
|Dialog.title custom action.chromium.png|1242| Changed|
|Dialog.title custom action - RTL.chromium.png|1242| Changed|
|Dialog.title custom action - High Contrast.chromium.png|3507| Changed|
|Dialog.scroll long content - RTL.chromium.png|611| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-dialog/src/contexts/dialogTransitionContext.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
  * @internal
  */
-export type DialogTransitionContextValue = TransitionStatus | undefined;
+export type DialogTransitionContextValue =
+  | 'entering'
+  | 'entered'
+  | 'idle'
+  | 'exiting'
+  | 'exited'
+  | 'unmounted'
```
Comment (layershifter) [inline packages/react-components/react-dialog/src/contexts/dialogTransitionContext.ts:12]: Inlined type to avoid dependency.

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31380&commitId=574f11046038eb43fe587e319b09e55e2dcce619&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|3| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by robertpenner — APPROVED** [review]
The motion logic is so concise and easy to read now. 🥇

File: packages/react-components/react-dialog/src/components/Dialog/renderDialog.tsx [hunk 3]
```diff
@@ -1,35 +1,26 @@
...
-            nodeRef={state.dialogRef}
-            appear={true}
-            // FIXME: this should not be hardcoded tokens.durationGentle
-            timeout={250}
-          >
-            {status => <DialogTransitionProvider value={status}>{content}</DialogTransitionProvider>}
-          </Transition>
+        {content && (
+          <DialogSurfaceMotion appear visible={open} unmountOnExit>
+            {content as React.ReactElement}
```
Comment (bsunderhus) [inline packages/react-components/react-dialog/src/components/Dialog/renderDialog.tsx:22]: nit: let's add a comment here on why this casting is ok
```suggestion
            // Casting here as content should be equivalent to <DialogSurface/>
            // FIXME: content should not be ReactNode it should be ReactElement instead.
            {content as React.ReactElement}
```

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts [hunk 4]
```diff
@@ -75,23 +75,24 @@ export const useDialogSurface_unstable = (
...
-    } else if (transitionStatus === 'exited') {
+
+    disableBodyScroll();
+
+    return () => {
       enableBodyScroll();
-    }
-  }, [disableBodyScroll, enableBodyScroll, isBodyScrollLocked, isNestedDialog, open, transitionStatus]);
+    };
+  }, [enableBodyScroll, isNestedDialog, disableBodyScroll, modalType]);
```
Comment (bsunderhus) [inline packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:89]: Would it be possible to move this whole logic back into the `useDialog` hook? it was originally there, it was only moved to `useDialogSurface` to get access to the `transitionStatus`.

**Review by bsunderhus — APPROVED** [review]

File: packages/react-components/react-drawer/src/components/OverlayDrawer/OverlayDrawerDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  type DialogContextValues,
+  type DialogProps,
+  DialogProvider,
+  type DialogState,
+  DialogSurfaceProvider,
+  useDialog_unstable,
+  useDialogContextValues_unstable,
+} from '@fluentui/react-dialog';
+
+const renderDialog_unstable = (state: DialogState, contextValues: DialogContextValues) => {
```
Comment (bsunderhus) [inline packages/react-components/react-drawer/src/components/OverlayDrawer/OverlayDrawerDialog.tsx:12]:
```suggestion
const renderOverlayDrawerDialog_unstable = (state: DialogState, contextValues: DialogContextValues) => {
```

File: change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31380&commitId=feced8cb74290edadd26c536c4faa13005f31a86&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|2| Changed|
|Avatar Converged.badgeMask.normal.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts (hunk 4 above)
Comment (layershifter) [reply packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:89]: As we discussed in offline, it makes sense to keep it there as we disable/enable scroll when `DialogSurface` gets mounted/unmounted.

File: change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json (hunk 1 above)
Comment (fabricteam) [inline change/@fluentui-react-dialog-6263971a-523d-4e53-9472-04268e11b2fc.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31380&commitId=ba6464970bfe99cdfeabcef0a2c415430f3950e4&env=prod&clientType=FLUENTUI)


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.spacing.default.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**